oomact_ros/src/CalibrationNode.cpp

```diff
@@ -5,98 +5,113 @@
+
+  ROS_INFO("Loading config file...");
 
   cal::ValueStoreRef vs = cal::ValueStoreRef::fromFile(config_file);
   cal::ValueStoreRef vs_model = vs.getChild("model");
 
-  ROS_INFO("Setting up model and trajectory");
+  ROS_INFO("Setting up model and trajectory...");
 
-  std::shared_ptr<cal::FrameGraphModel> model =
-      std::make_shared<cal::FrameGraphModel>(vs_model);
+  std::shared_ptr<cal::FrameGraphModel> model = std::make_shared<cal::FrameGraphModel>(vs_model);
   cal::PoseTrajectory traj(*model, "traj", vs_model);
   model->add(traj);
 
-  ROS_INFO("Loading pose sensor parameters");
-
-  int num_pose_sensors;
-  nh_private.param("num_pose_sensors", num_pose_sensors, 0);
-  if (num_pose_sensors < 0) {
-    ROS_ERROR("Number of pose sensors cannot be negative");
-    num_pose_sensors = 0;
-  }
+  ROS_INFO("Loading sensor parameters...");
 
-  std::vector<cal::PoseSensor> pose_sensors;
-  pose_sensors.reserve(num_pose_sensors);
-  for (size_t i = 0; i < num_pose_sensors; ++i) {
-    pose_sensors.emplace_back(*model, std::string("pose") + std::to_string(i),
-                              vs_model);
-    model->add(pose_sensors.back());
-  }
+  std::vector<std::unique_ptr<cal::Sensor>> sensors;
+  loadSensorParameters(vs.getChild("sensors"), model, &sensors);
 
-  ROS_INFO_STREAM("Loaded " << pose_sensors.size() << " pose sensors");
+  ROS_INFO_STREAM("Loaded " << sensors.size() << " sensors.");
 
-  ROS_INFO("Setting up calibration");
+  ROS_INFO("Setting up calibration...");
 
   model->init();
 
-  std::unique_ptr<cal::BatchCalibratorI> calibrator =
-      cal::createBatchCalibrator(vs.getChild("calibrator"), model);
-
-  ROS_INFO("Loading data bag");
+  std::unique_ptr<cal::BatchCalibratorI> calibrator = cal::createBatchCalibrator(vs.getChild("calibrator"), model);
 
-  std::string bag_file;
-  if (!nh_private.getParam("bag_file", bag_file)) {
-    ROS_FATAL("Could not find bag_file parameter, exiting");
-    exit(EXIT_FAILURE);
-  }
+  ROS_INFO("Loading data bag...");
 
   cal::ros::RosInputProvider(model).feedBag(bag_file, *calibrator);
 
   ROS_INFO("Starting calibration...");
 
   calibrator->calibrate();
 
-  ROS_INFO("Calibration done, generating results");
+  ROS_INFO("Calibration done, generating results...");
 
-  // Output results
-  for (size_t i = 0; i < pose_sensors.size(); ++i) {
-    ROS_INFO_STREAM("Pose sensor  : " << i);
-    ROS_INFO_STREAM("  Translation: "
-                    << pose_sensors[i].getTranslationToParent().transpose());
-    ROS_INFO_STREAM(
-        "  Rotation   : "
-        << pose_sensors[i].getRotationQuaternionToParent().transpose());
+  for (const std::unique_ptr<cal::Sensor>& sensor : sensors) {
+    ROS_INFO_STREAM("Sensor       : " << sensor->getName());
+    ROS_INFO_STREAM("  Translation: " << sensor->getTranslationToParent().transpose());
+    ROS_INFO_STREAM("  Rotation   : " << sensor->getRotationQuaternionToParent().transpose());
```

Comment:
This is one of the big questions: which transformation to provide for complexer multi body systems. What you have here is just  w.r.t. to the parent frame.